master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/ha/StateMachine.java

```diff
@@ -344,6 +344,7 @@ public long takeSnapshot() {
         LOG.warn("Failed to rename snapshot from {} to {}.", tempFile, snapshotFile);
         return RaftLog.INVALID_LOG_INDEX;
       }
+      storage.loadLatestSnapshot();
```

Comment:
Good catch! GA failed only for spark-3.1, can you re-trigger again?

Reply:
Maybe we should remove this method to `StateMachine`'s `getLatestSnapshot` with a try catch.

Reply:
`StateMachine`'s `getLatestSnapshot` is called every turn when `LogAppender` sends `AppendEntries`, see https://github.com/apache/ratis/blob/729d3dce220f352a720cbe38d0762399b6e55f06/ratis-grpc/src/main/java/org/apache/ratis/grpc/server/GrpcLogAppender.java#L154.
`loadSnapshot` is a heavy operation, since it will scan the file system and uses regular expressions. If we move this method to `getLatestSnapshot`, it maybe a performance problem. What do you think?

Reply:
Yea， I have just thought about such a thing too...